float64 x
float64 y
int32 frame_x
int32 frame_y
